test/com/sun/jna/DirectTest.java

```diff
@@ -202,8 +204,8 @@ public void testGetOptionsForDirectMappingWithMemberInitializer() {
         };
         final TypeMapper mapper = new DefaultTypeMapper();
         final int alignment = Structure.ALIGN_NONE;
-        final String encoding = System.getProperty("file.encoding");
-        Map<String, Object> options = new HashMap<String, Object>();
+        final String encoding = Charset.defaultCharset().displayName();
```

Comment:
Not sure here, but the display name, that is even localized, seems wrong. The canonical name of the charset seems to be a better fit:

```suggestion
        final String encoding = Charset.defaultCharset().name();
```

Same would apply in line 228